pkg/events/events.go

```diff
@@ -124,9 +125,9 @@ func New(object interface{}, eventType config.EventType, resource, clusterName s
 		event.Action = eventObj.Action
 		event.TimeStamp = eventObj.LastTimestamp.Time
 		// Compatible with events.k8s.io/v1
-		if eventObj.LastTimestamp.IsZero() {
-			event.TimeStamp = eventObj.Series.LastObservedTime.Time
-			event.Count = eventObj.Series.Count
+		if eventObj.LastTimestamp.IsZero() && !eventSeriesObj.LastObservedTime.IsZero() {
```

Comment:
Shouldn't we be using `eventObj.Series` instead of `eventSeriesObj`? Where are we setting `eventSeriesObj`? Is it always going to be nil?

Reply:
you will get null pointer there, because series object is not part of coreV1.Event

https://pkg.go.dev/k8s.io/api/core/v1#EventSeries

https://pkg.go.dev/k8s.io/api/core/v1#Event

when we are trying to use event.series its always giving nil pointer

Reply:
// Data about the Event series this event represents or nil if it's a singleton Event.
	// +optional
	Series *[EventSeries](https://pkg.go.dev/k8s.io/api/core/v1#EventSeries) `json:"series,omitempty" protobuf:"bytes,11,opt,name=series"`